### PR TITLE
Remove outdated Kuryr + Octavia compatibility notes

### LINKED ISSUE #23711: [enterprise-4.5] Update Kuryr and Octavia compatibility statements in OSP installation docs
### Which section(s) is the issue in?

From @luis5tb:

>I just saw the new docs for 4.5 and saw something that must be removed in here: https://docs.openshift.com/container-platform/4.5/installing/installing_openstack/installing-openstack-installer-kuryr.html#installation-osp-kuryr-known-limitations_installing-openstack-installer-kuryr
>the next is no longer true: "	
>The OVN Octavia driver does not support listeners that use different protocols on any RHOSP version."
>so, it can be removed
>and similarly the second bullet at RHOSP environment limitations: "The OVN Octavia driver is used"

### What needs fixing?

These limitations are no longer true for the latest release. They should be removed.

### PR DESCRIPTION
**Pull request by maxwelldb**
This PR resolves #23711. 

QE should be covered already by https://github.com/openshift/openshift-docs/pull/21488 and https://github.com/openshift/openshift-docs/pull/22878.

### CODE REVIEW

**Comment by maxwelldb** [comment]
Let me know if you have any concerns, @luis5tb.

**Comment by maxwelldb** [comment]
@openshift/team-documentation PTAL. S review: +1 and -11.

File: modules/installation-osp-kuryr-known-limitations.adoc [hunk 1]
```diff
@@ -34,23 +34,13 @@ OpenShift UDP services are not supported.
...
 [id="openstack-go-limitations_{context}"]
 == {rh-openstack} environment limitations
 
 There are limitations when using Kuryr SDN that depend on your deployment environment.
 
-Because of Octavia's lack of support for the UDP protocol and multiple listeners, Kuryr forces Pods to use TCP
-for DNS resolution if:
-
-* The {rh-openstack} version is earlier than 16
-* The OVN Octavia driver is used
```
Comment (maxwelldb) [inline modules/installation-osp-kuryr-known-limitations.adoc:53]: This bullet no longer applies--OVN Octavia support is improved in 4.5+.

File: modules/installation-osp-kuryr-known-limitations.adoc [hunk 2]
```diff
@@ -34,23 +34,13 @@ OpenShift UDP services are not supported.
 same port. Services that expose the same port to different protocols, like TCP
 and UDP, are not supported.
 
-[IMPORTANT]
```
Comment (maxwelldb) [inline modules/installation-osp-kuryr-known-limitations.adoc:37]: This admonition no longer applies--OVN Octavia support is improved in 4.5+.

**Review by adellape — APPROVED** [review]
:+1:

**Comment by maxwelldb** [comment]
@adellape Thanks!

**Pull request merged**